android/src/main/java/io/customer/reactnative/sdk/messagingpush/RNCIOPushMessaging.kt

```diff
@@ -0,0 +1,104 @@
+     * has been completed.
+     *
+     * @param pushConfigurationOptions configurations options for push notifications, required for
+     * iOS only, unused on Android.
+     * @param promise to resolve and return the results.
+     */
+    @ReactMethod
+    fun showPromptForPushNotifications(pushConfigurationOptions: ReadableMap?, promise: Promise) {
+        // Skip requesting permissions for older versions and when permission already granted
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.TIRAMISU || checkPushPermissionStatus() == PermissionStatus.Granted) {
+            promise.resolve(PermissionStatus.Granted.toReactNativeResult)
+            return
+        }
+
+        try {
+            notificationRequestPromise = promise
+            (currentActivity as PermissionAwareActivity).requestPermissions(
+                arrayOf(Manifest.permission.POST_NOTIFICATIONS),
+                POST_NOTIFICATIONS_PERMISSION_REQUEST,
+                this,
+            )
+        } catch (ex: Throwable) {
+            promise.reject(ex)
+            notificationRequestPromise = null
+        }
+    }
+
+    /**
+     * Checks current permission of push notification permission
+     */
+    private fun checkPushPermissionStatus(): PermissionStatus =
+        if (ContextCompat.checkSelfPermission(
+                reactContext,
+                Manifest.permission.POST_NOTIFICATIONS,
+            ) == PackageManager.PERMISSION_GRANTED
+        ) PermissionStatus.Granted
+        else PermissionStatus.Denied
+
+    /**
+     * Resolves and clears promise with the provided permission status
+     */
+    private fun resolvePermissionPromise(status: PermissionStatus) {
+        notificationRequestPromise?.resolve(status.toReactNativeResult)
+        notificationRequestPromise = null
+    }
+
+    override fun onRequestPermissionsResult(
+        requestCode: Int, permissions: Array<String>, grantResults: IntArray,
+    ): Boolean = when (requestCode) {
+        POST_NOTIFICATIONS_PERMISSION_REQUEST -> {
+            // If request is cancelled, the result arrays are empty.
+            if (grantResults.isNotEmpty() && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                resolvePermissionPromise(PermissionStatus.Granted)
+            } else {
+                resolvePermissionPromise(PermissionStatus.Denied)
+            }
+            true // as this permission can be removed now
+        }
+        else -> false // desired permission not yet granted, so we will keep the listener
+    }
+
+    override fun getName(): String = "CustomerioPushMessaging"
+
+    /**
+     * Maps native class to react native supported type so the result can be passed on to JS/TS classes.
+     */
+    private val PermissionStatus.toReactNativeResult: Any
+        get() = name
+
+    companion object {
+        private const val POST_NOTIFICATIONS_PERMISSION_REQUEST = 24676
```

Comment:
I am not sure what `24676` mean?

Reply:
Just a random number. It should be a unique number which Android OS returns when permission is granted/denies so the caller can be identified, can be any number 😄